pkg/util/timeutil/timer.go

```diff
@@ -79,8 +79,14 @@ func (t *Timer) Reset(d time.Duration) {
 		t.C = t.timer.C
 		return
 	}
-	if !t.timer.Stop() && !t.Read {
-		<-t.C
+	// Stop the timer and, if the timer had fired already, attempt to drain its
+	// channel in case the user had not already received from the channel before
+	// calling Reset().
+	if !t.timer.Stop() {
+		select {
+		case <-t.C:
```

Comment:
How is this working if the event has not arrived yet? (i.e. the deadline has not yet expired)